p2p/pex/reactor.go

```diff
@@ -81,25 +84,22 @@ func (r *ReactorV2) handlePexMessage(envelope p2p.Envelope) error {
 	// only processing addresses we actually requested.
 	switch msg := envelope.Message.(type) {
 	case *protop2p.PexRequest:
-		endpoints := r.peerManager.Advertise(envelope.From, maxAddresses)
-		resp := &protop2p.PexResponse{Addresses: make([]protop2p.PexAddress, 0, len(endpoints))}
-		for _, endpoint := range endpoints {
-			// FIXME: This shouldn't rely on NetAddress.
-			resp.Addresses = append(resp.Addresses, endpoint.NetAddress().ToProto())
+		pexAddresses := r.resolve(r.peerManager.Advertise(envelope.From, maxAddresses), maxAddresses)
```

Comment:
Do we need to have `maxAddresses` passed to `resolve` since `r.peerManager.Advertise` returns at most `maxAddresses`?

Reply:
Yes, because an URL may resolve to many IP addresses.

Reply:
But we do a `if len(endpoints) >= int(limit) { // ... }` check.

Reply:
Oh, that check is wrong. It should be `len(pexAddresses)`, not `len(endpoints)`. But in any case, `limit` in that check comes from `maxAddresses`.